# Part -1 Python Basics (Variables)

1. Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [2]:
print("Name: Taha\nFather's Name: Amir\nDate of Birth: 29-12-24")


Name: Taha
Father's Name: Amir
Date of Birth: 29-12-24


2. Write your small bio using variables and print it using print function

In [3]:
name="taha"
father_name="amir"
age="18"
date_of_birth= "29-12-2005"
address="liaquatabad karachi"
print(f"hi my name is {name} and my father name is {father_name} , my date of birth is {date_of_birth} so my age is {age} my address is {address}")

hi my name is taha and my father name is amir , my date of birth is 29-12-2005 so my age is 18 my address is liaquatabad karachi


3. Write a program in which use all the operators we can use in Python

In [4]:
#basic mathematics
a=10
b=15
print(f" addition {a+b}")
print(f" subtraction {a-b}")
print(f" multiplication {a*b}")
print(f" division {a/b}")

#Assignment operators

x=10
x+=5
print(f"x is equal to {x}")

#comparision operators

a=10
b=5
print(f"a is equal to b?{a==b}")
print(f"a is not equal to b?{a!=b}")
print(f"a is greater then b?{a>b}")
print(f"a is not greater then b?{b>a}")

#logical operators
a=True
b=False
print(f"a AND b:{a and b}")
print(f"a OR b:{a or b}")
print(f" NOT a:{ not a}")

 addition 25
 subtraction -5
 multiplication 150
 division 0.6666666666666666
x is equal to 15
a is equal to b?False
a is not equal to b?True
a is greater then b?True
a is not greater then b?False
a AND b:False
a OR b:True
 NOT a:False


4. Completes the following steps of small task:
    - Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
    - Mention Variable of Total Marks and assign 300 to it
    - Calculate Percentage

In [5]:
eng=90
math=84
islamiat=88
total=300
sum=eng+math+islamiat
print(f"your percentage is {sum/total*100}")

your percentage is 87.33333333333333


# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [1]:
service=int(input("enter your years of service:"))
salary=int(input("enter your salary:"))
if service>5:
    print(f"your bonus salary is {salary*0.05}")
else:
    print("no incremenet")

enter your years of service:4
enter your salary:39000
no incremenet



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [2]:
age=int(input("enter your age:"))
if age >=17:
    print("you are elegible for voting")
else:
    print("you re not elegible")

enter your age:56
you are elegible for voting



3) Write a program to check whether a number entered by user is even or odd.

In [3]:
number=int(input("enter your number:"))
if number %2 == 0:
    print("number is even")
else:
    print("number is odd") 

enter your number:4
number is even


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [4]:
number=int(input("enter your number:"))
if number %7 == 0:
    print("number is divisible by 7")
else:
    print("number is not divisible by 7") 

enter your number:49
number is divisible by 7


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [5]:
number=int(input("enter your number:"))
if number %5 == 0:
    print("hello")
else:
    print("bye") 

enter your number:45
hello


7) Write a program to display the last digit of a number.


In [9]:
number = int(input("Enter a number: "))

if number < 0:
    last_digit = number % 10
if last_digit == 0:
    print("The last digit is 0")
elif last_digit == 1:
    print("The last digit is 1")
elif last_digit == 2:
    print("The last digit is 2")
elif last_digit == 3:
    print("The last digit is 3")
elif last_digit == 4:
    print("The last digit is 4")
elif last_digit == 5:
    print("The last digit is 5")
elif last_digit == 6:
    print("The last digit is 6")
elif last_digit == 7:
    print("The last digit is 7")
elif last_digit == 8:
    print("The last digit is 8")
elif last_digit == 9:
    print("The last digit is 9")
else:
    print("Error")  

Enter a number: 4567


NameError: name 'last_digit' is not defined

9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [10]:
length=int(input("enter value of length:"))
breadth=int(input("enter value of breadth:"))
if length==breadth:
    print("square")
else:
    print("rectangle")
           

enter value of length:4
enter value of breadth:5
rectangle


10) Take two int values from user and print greatest among them.

In [11]:
value_1=int(input("enter your value:"))
value_2=int(input("enter your value:"))
if value_1>value_2 :
    print(f"{value_1}")
elif value_2>value_1 :
    print(f"{value_2}")

enter your value:4
enter your value:6
6


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [12]:
quantity = int(input("Enter the quantity of units you want to purchase: "))
unit_cost = 100
total_cost = quantity * unit_cost
if total_cost > 1000:
    discount = total_cost * 0.10
    total_cost -= discount
print("The total cost is:", total_cost)

Enter the quantity of units you want to purchase: 45
The total cost is: 4050.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [23]:
marks=int(input("enter your marks:"))
if marks <= 25:
    print("ypur grade is F")
    
elif marks >= 25 and marks < 45:
    print("your grade is E")
elif marks >= 45 and marks < 50:
    print("yur grade is D")
elif marks >= 50 and marks < 60:
    print("your grade is C")
elif marks >= 60 and marks < 80:
    print("your grade is B")
elif marks >= 80:
    print("your grade is A")
else:
    print("invalid marks entered")



enter your marks:45
yur grade is D


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [24]:
classes=int(input("enter num of classes held:"))
attendence=int(input("enter m of classes attend:"))
if attendence<0.075:
    print(" u will not be allowed to sit in exam")
else:
    print(" u will allow to sit in exam")

enter num of classes held:65
enter m of classes attend:45
 u will allow to sit in exam


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [25]:
medical=input("enter your medical cause(Y/N):")
if medical=="Y":
    print("u re allow to sit in exam")
else:
    print("u re not allow to sit in exam ")

enter your medical cause(Y/N):N
u re not allow to sit in exam 


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [26]:
year = int(input("Enter a year: "))
if year % 4 == 0:
    print(f"{year} is a leap year.")
elif year % 100 == 0:
    print(f"{year} is a leap year.")
elif year % 400 == 0:
    print(f"{year} is not a leap year.")
else:
     print(f"{year} is not a leap year.")

Enter a year: 2019
2019 is not a leap year.


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [28]:
gender=input("enter your gender (M/F)")
age=int(input("enter your age:"))
martial_status=input("enter your martial status(Y/N)")
if gender=="F":
    print("u work in urban areas")
if gender=="M":  
    if age<40>20:
        print("u will wok anywhere")
if gender=="M":
    if age<60>40:
        print("u will work in urban areas")
else:
    print("error")

enter your gender (M/F)M
enter your age:34
enter your martial status(Y/N)Y
u will wok anywhere
u will work in urban areas


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [31]:
#6
unit=int(input("enter your units:"))
if unit<100:
    print("your bill amount is 0")
elif unit<200>100:
    print(f"your bill amount is {unit*5}")
elif unit<400>200:
     print(f"your bill amount is {unit*10}")


enter your units:450


In [32]:
#13
age1 = int(input("Enter the age of person 1: "))
age2 = int(input("Enter the age of person 2: "))
age3 = int(input("Enter the age of person 3: "))
oldest = max(age1, age2, age3)
youngest = min(age1, age2, age3)
print(f"The oldest person is {oldest} years old.")
print(f"The youngest person is {youngest} years old.")

Enter the age of person 1: 45
Enter the age of person 2: 54
Enter the age of person 3: 34
The oldest person is 54 years old.
The youngest person is 34 years old.
